test/plugin/index.spec.js

```diff
@@ -24,4 +27,70 @@ describe("plugin", () => {
     // eslint-disable-next-line no-magic-numbers
     expect(pluginWithOptions.port).to.equal(3000);
   });
+
+  describe("plugin methods", () => {
+    let stats;
+    let toJson;
+    let compilation;
+    let compiler;
+    beforeEach(() => {
+      stats = {};
+      toJson = base.sandbox.stub().returns(stats);
+      compilation = {
+        errors: [],
+        warnings: [],
+        getStats: () => ({ toJson }),
+        tap: base.sandbox.stub(),
+        tapAsync: base.sandbox.stub() // this is us in webpack-dashboard
+      };
+      compiler = {
+        // TODO: ONLY WEBPACK4, but that's what we have in devDeps
+        hooks: {
+          compilation,
+          emit: {
+            intercept: base.sandbox.stub()
+          },
+          watchRun: {
+            tapAsync: base.sandbox.stub()
+          },
+          run: {
+            tapAsync: base.sandbox.stub()
+          },
+          compile: {
+            tap: base.sandbox.stub()
+          },
+          failed: {
+            tap: base.sandbox.stub()
+          },
+          invalid: {
+            tap: base.sandbox.stub()
+          },
+          done: {
+            tap: base.sandbox.stub()
+          }
+        }
+      };
+    });
+
+    it("can do a basic compilation", () => {
+      const plugin = new Plugin();
+      expect(plugin.apply(compiler)).to.not.throw;
+
+      // after instantiation, test that we can hit observeMetrics
+      expect(plugin.observeMetrics({ toJson })).to.not.throw;
+    });
+
+    it("can do a basic observeMetrics", () => {
+      const plugin = new Plugin();
+      const actions = sinon.spy(inspectpack, "actions");
+
+      plugin.observeMetrics({ toJson }).subscribe({
+        next: () => {},
+        error: () => {},
```

Comment:
GLOBAL TIP: Use `base.sandbox.spy()` in place of `() => {}`. It can be asserted on and it's more clear your using a test double.